src/etc/inc/auth.inc

```diff
@@ -474,6 +474,8 @@ function local_user_set(& $user) {
 		$user_shell = "/bin/tcsh";
 	} elseif (userHasPrivilege($user, "user-copy-files")) {
 		$user_shell = "/usr/local/bin/scponly";
+	} elseif (userHasPrivilege($user, "user-copy-files-chroot")) {
```

Comment:
I'm thinking maybe this check should be above the test for user-copy-files, that way if the user somehow has both set, the more secure option of the two is used. That way they could be in a group with the weaker privilege but have the stricter privilege applied.